python/lsst/ts/externalscripts/base_make_bias.py

```diff
@@ -0,0 +1,227 @@
+                images_remaining -= 1
+                counter = 0
+            if counter == max_counter:
+                self.log.info(f"WARN: Maximum number of loops ({max_counter}) reached while waiting "
+                              "for image in archiver")
+                break
+            counter += 1
+
+        self.ocps.evt_job_result.flush()
+
+        # Now run the bias pipetask via the OCPS
+        ack = await self.ocps.cmd_execute.set_start(
+            wait_done=False, pipeline="${CP_PIPE_DIR}/pipelines/cpBias.yaml", version="",
+            config=f"-j 8 -i {self.config.input_collections} --register-dataset-types -c isr:doDefect=False "
+                   f"-c isr:doLinearize=False -c isr:doCrosstalk=False "
+                   f"-c isr:overscan.fitType='MEDIAN_PER_ROW'",
+            data_query=f"instrument='{self.instrument_name}' AND"
+                       f" detector IN {self.config.detectors} AND exposure IN {exposures}"
+        )
+        if ack.ack != salobj.SalRetCode.CMD_ACK:
+            ack.print_vars()
+
+        # Wait for the in-progress acknowledgement with the job identifier.
+        ack = await self.ocps.cmd_execute.next_ackcmd(ack, wait_done=False)
+        self.log.debug('Received acknowledgement of ocps command')
+
+        ack.print_vars()
+        job_id = json.loads(ack.result)["job_id"]
+
+        # Wait for the command completion acknowledgement.
+        ack = await self.ocps.cmd_execute.next_ackcmd(ack)
+        self.log.debug('Received command completion acknowledgement from ocps')
+        if ack.ack != salobj.SalRetCode.CMD_COMPLETE:
+            ack.print_vars()
+        # Wait for the job result message that matches the job id we're
+        # interested in ignoring any others (from other remotes).
+        # This obviously needs to follow the first acknowledgement
+        # (that returns the, job id) but might as well wait for the second.
+        while True:
+            msg = await self.ocps.evt_job_result.next(flush=False, timeout=600)
+            response = json.loads(msg.result)
+            if response["jobId"] == job_id:
+                break
+
+        self.log.info(f"Final status: {response}")
+
+        # Certify the bias, if the job completed successfully
+        if not response['phase'] == 'completed':
+            raise RuntimeError(f"Bias creation not completed successfully: {response['phase']}")
+        else:
+            self.log.info("Certifying bias: ")
+            # certify the bias
+            REPO = self.config.repo
+            # This is the output collection where the OCPS puts the biases
+            BIAS_DIR = f"u/ocps/{job_id}"
+            CAL_DIR = self.config.calib_dir
+            cmd = (f"butler certify-calibrations {REPO} {BIAS_DIR} {CAL_DIR} " 
+                    "--begin-date 1980-01-01 --end-date 2050-01-01 bias")
+            self.log.info(cmd)
+            try:
+                os.system(cmd)
```

Comment:
uhm.. using `os.system` is probably not the best idea here. If this call blocks it will seem like the Script died (it will stop sending heartbeats) and it will also become unresponsive to DDS commands... 

Asyncio has a utility function to run "system calls" without blocking the event loop. 

Use:

```
process = await asyncio.create_subprocess_shell(cmd)

stdout, stderr = await process.communicate()

self.log.debug(f"Process returned: {process.returncode}")
```

you probably want to raise an exception if the return code is different than zero. 

You can also log the output if you want, probably if there's an error:

```
self.log.debug(stdout)
self.log.error(stderr)
```

Reply:
Another thing to consider here, Is this butler instance accessible at the summit and/or script queue?

Reply:
I'm not sure I understand, my apologies. When running this script from a notebook at {{summit-lsp.lsst.codes}} the DM stack needs to be loaded first via something like {{source $LOADSTACK }}, for example, so that {{butler certify}}  is available.